api/authentication/base.py

```diff
@@ -0,0 +1,532 @@
+        be associated with circulation events -- but a circulation
+        event is not associated with the patron who triggered it.
+
+        :param cached_neighborhood: This is the same as neighborhood,
+        but it _will_ be cached in the patron's database record, for
+        up to twelve hours. This should only be used by ILS systems
+        that would have performance problems fetching patron
+        neighborhood on demand.
+
+        If cached_neighborhood is set but neighborhood is not,
+        cached_neighborhood will be used as neighborhood.
+
+        :param complete: Does this PatronData represent the most
+        complete data we are likely to get for this patron from this
+        data source, or is it an abbreviated version of more complete
+        data we could get some other way?
+        """
+        self.permanent_id = permanent_id
+
+        self.set_authorization_identifier(authorization_identifier)
+        self.username = username
+        self.authorization_expires = authorization_expires
+        self.external_type = external_type
+        self.fines = fines
+        self.block_reason = block_reason
+        self.library_identifier = library_identifier
+        self.complete = complete
+
+        # We do not store personal_name in the database, but we provide
+        # it to the client if possible.
+        self.personal_name = personal_name
+
+        # We do not store email address in the database, but we need
+        # to have it available for notifications.
+        self.email_address = email_address
+
+        # If cached_neighborhood (cached in the database) is provided
+        # but neighborhood (destroyed at end of request) is not, use
+        # cached_neighborhood as neighborhood.
+        self.neighborhood = neighborhood or cached_neighborhood
+        self.cached_neighborhood = cached_neighborhood
+
+    def __eq__(self, other):
+        """
+        Compares two PatronData objects
+
+        :param other: PatronData object
+        :type other: PatronData
+
+        :return: Boolean value indicating whether two items are equal
+        :rtype: bool
+        """
+
+        if not isinstance(other, PatronData):
+            return False
+
+        return (
+            self.permanent_id == other.permanent_id
+            and self.username == other.username
+            and self.authorization_expires == other.authorization_expires
+            and self.external_type == other.external_type
+            and self.fines == other.fines
+            and self.block_reason == other.block_reason
+            and self.library_identifier == other.library_identifier
+            and self.complete == other.complete
+            and self.personal_name == other.personal_name
+            and self.email_address == other.email_address
+            and self.neighborhood == other.neighborhood
+            and self.cached_neighborhood == other.cached_neighborhood
+        )
```

Comment:
Do we have a test that covers this `__eq__`? I couldn't see one.
That's the kind of code that's easy to break without realizing, and causes next-to-impossible-to-locate problems.

Reply:
It doesn't look like we have a test for this. I think it would be worth adding one.

I don't want to do that as part of this PR though. This is literally just a copy and paste to move classes around into other files, and update the import paths, no other changes are included. The code is exactly the same as it was before. 

I wanted to move the files around this way, so its easier to see what actually changed in #1074.